actions/subscribers.go

```diff
@@ -370,16 +406,42 @@ func ImportSubscribers(c *gin.Context) {
 		return
 	}
 
-	go func(ctx context.Context, client s3iface.S3API, filename string, userID int64, segs []entities.Segment) {
+	res, err := client.GetObject(&s3.GetObjectInput{
+		Bucket: aws.String(os.Getenv("FILES_BUCKET")),
+		Key:    aws.String(fmt.Sprintf("subscribers/import/%d/%s", u.ID, body.Filename)),
+	})
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
+			"message": "Unable to import subscribers. Please try again.",
+		})
+		return
+	}
+
+	csvCount, err := utils.CSVLineCounter(res.Body)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
+			"message": "Unable to import subscribers. Please try again.",
+		})
+	}
+
+	if count+int64(csvCount) > u.Boundaries.SubscribersLimit {
+		c.JSON(http.StatusForbidden, gin.H{
+			"message":            "You will have exceeded your subscribers limit with this import, please upgrade to a bigger plan or contact support.",
+			"your_limit":         count,
+			"to_be_exceeded_for": count + int64(csvCount) - u.Boundaries.SubscribersLimit,
+		})
+	}
+
+	go func(ctx context.Context, user *entities.User, segs []entities.Segment, res *s3.GetObjectOutput) {
 		imp := importer.NewS3SubscribersImporter(client)
```

Comment:
@FilipNikolovski @Dzalevski I am not sure this is more like a question. Can the client here make a problem?

Reply:
How @gudgl

Reply:
We can pass it down as a dependency, but I don't think it will make a problem.